backend/apid/middlewares/authentication.go

```diff
@@ -75,43 +77,32 @@ func (a Authentication) Then(next http.Handler) http.Handler {
 
 func extractAPIKeyClaims(ctx context.Context, key string, store storev2.Interface) (*corev2.Claims, error) {
 	var claims *corev2.Claims
-	// retrieve the APIKey based on the key provided
-	apiKey := &corev2.APIKey{
-		ObjectMeta: corev2.ObjectMeta{
-			Name: key,
-		},
-	}
-	req := storev2.NewResourceRequestFromResource(apiKey)
-	wrapper, err := store.GetConfigStore().Get(ctx, req)
+	keyStore := storev2.Of[*corev2.APIKey](store)
+	apiKeys, err := keyStore.List(ctx, storev2.ID{}, nil)
 	if err != nil {
 		return nil, err
 	}
-	if err := wrapper.UnwrapInto(apiKey); err != nil {
-		return nil, err
-	}
 
-	user := &corev2.User{
-		Username: apiKey.Username,
-	}
-
-	req = storev2.NewResourceRequestFromResource(user)
-	wrapper, err = store.GetConfigStore().Get(ctx, req)
-	if err != nil {
-		return nil, err
-	}
+	for _, apiKey := range apiKeys {
```

Comment:
Suppose the way bcrpyt works we can't actually look up a particular hash, huh?

You worried at all about this becoming a bourdon in highly automated environments?

Reply:
That's correct. Since salt selection is internal to how bcrypt works, we can't hash the password and get the same result. I had forgotten about that fact going into this.

If we could select our own salts with another algorithm like scrypt then we would have the ability to get the same hash. But that in itself would weaken our security posture, because you are supposed to use a completely random salt.

We could improve lookup efficiency by having the client include the user the API key is associated with, which is an additional client burden, but could reduce server-side costs a lot. For now, I'm tempted to wait and see if it becomes a problem.

Reply:
I suppose yet another trick we could pull is to include both the username and the API key in an encoded payload and hand _those_ out instead of UUIDs. Then we'd be able to perfectly query the set of API keys for a given user, and the client experience would be the same.

Reply:
We can improve the efficiency of lookups in a future effort.